include/glow/Graph/Graph.h

```diff
@@ -270,7 +270,7 @@ class Function final : public Named {
   LogNode *createLog(llvm::StringRef name, NodeValue input);
 
   SoftMaxNode *createSoftMax(llvm::StringRef name, NodeValue input,
```

Comment:
@tlepley-cadence @rdzhabarov Most of our builders overload two versions of the builders: one with outTy and one without. Making outTy optional could be a good idea, but we need to be consistent. Not sure how to enforce the mandatory out type in a sane way.

Reply:
@nadavrot, I agree, let's do this consistently. I think there are two cases:
* Output shape calculations are based on the input shapes, and we do that inside the method which does not take outTy explicitly. The one which takes outTy just do the verification.
* We simply pass outTy as input.getType() to the other method

For the second case, we can eliminate one of the overloads and just check outTy for nullability. For the first case, we'd need to keep two overloads.

I think the way it's handled in the PR is the right way. Once this is merged I'll make sure we follow this practice elsewhere in the Graph.(h|cpp).

Reply:
Ok.